src/components/OnchainName.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useMemo } from 'react';
+import { getSlicedAddress } from '../core/address';
+import { useEnsName } from '../hooks/useEnsName';
+import type { Address } from 'viem';
+
+type OnchainNameProps = {
+  address?: Address;
```

Comment:
I think the `address` should be mandatory

Reply:
agreed with @cnasc. vote to accept to make less cumbersome for consumers.